src/gt4py/gtc/utils.py

```diff
@@ -0,0 +1,53 @@
+import collections.abc
+
+from devtools import debug  # noqa: F401
+
+
+def flatten_list(nested_iterables, filter_none=False, *, skip_types=(str, bytes)):
+    return list(flatten_list_iter(nested_iterables, filter_none, skip_types=skip_types))
+
+
+def flatten_list_iter(nested_iterables, filter_none=False, *, skip_types=(str, bytes)):
+    for item in nested_iterables:
+        if isinstance(item, list) and not isinstance(item, skip_types):
+            yield from flatten_list(item, filter_none)
+        else:
+            if item is not None or not filter_none:
+                yield item
+
+
+class ListTuple(collections.abc.Sequence):
+    def __init__(self, *args):
+        assert len(args) > 0
+        assert all([isinstance(arg, list) for arg in args])
+        self.tpl = args
+
+    def __len__(self):
+        return len(self.tpl)
+
+    def __getitem__(self, key):
+        return self.tpl[key]
+
+    def __add__(self, other):
```

Comment:
This function has a lot of if/else branches!

Reply:
I would suggest splitting or at least commenting.

Reply:
done